### PR TITLE
Use megaparsec instead of attoparsec to get better error messages

### PR DESCRIPTION
**Pull request by anka-213**
There are currently a few hacks to make it work. I will clean up the code soon. I should also add some example of how the error messages are better.

### CODE REVIEW

**Comment by anka-213** [comment]
I also have commented out a test, since I was unsure of how to translate it. Will fix that too soon.

File: languages/haskell/app/Main.hs [hunk 1]
```diff
@@ -184,7 +187,7 @@ getTableLine = do
 
 irrelevantLine :: Parser (Maybe String)
 irrelevantLine = do
-  endOfLine <|> (notChar '|' >> skipWhile (/= '\n') >> endOfLine)
+  endOfLine <|> (notChar '|' >> skipWhile "character" (/= '\n') >> endOfLine)
```
Comment (mengwong) [inline languages/haskell/app/Main.hs:190]: O_o

**Comment by mengwong** [comment]
Looks like there's a tiny bit of cross-contamination coming from the XML side of things, but the conversion from attoparsec to megaparsec is done! You did it! Yay!

**Comment by anka-213** [comment]
Yes, that cross-contamination was intentional to avoid merge conflicts, but maybe it would have been better to just accept the merge conflicts so we can avoid contamination.

File: languages/haskell/app/Main.hs (hunk 1 above)
Comment (anka-213) [reply languages/haskell/app/Main.hs:190]: "character" is the label for the individual characters in the line. Megaparsec's combinator that corresponds to `skipWhile` ([`takeWhileP`](https://hackage.haskell.org/package/megaparsec-8.0.0/docs/Text-Megaparsec.html#v:takeWhileP)) allows setting a label for what your condition corresponds to.

I am not entirely sure what that line is supposed to do though? Does it skip any line that does not start with `|`?

**Pull request merged**